includes/cli/class-data-backfill.php

```diff
@@ -0,0 +1,362 @@
+			}
+		}
+	}
+
+	/**
+	 * Process a WooCommerce entity.
+	 *
+	 * @param array  $data The data.
+	 * @param int    $timestamp The timestamp.
+	 * @param string $action The action.
+	 */
+	private static function process_event_entity( $data, $timestamp, $action ) {
+		if ( Site_Role::is_hub() ) {
+			$event = new \Newspack_Network\Incoming_Events\Abstract_Incoming_Event(
+				get_bloginfo( 'url' ),
+				$data,
+				$timestamp,
+				$action
+			);
+			$event->process_in_hub();
+			self::$results[ $action ][ $event->is_persisted ? 'processed' : 'duplicate' ]++;
+		} else {
+			$requests = self::find_webhook_requests( $action, $timestamp, $data );
+			if ( count( $requests ) > 0 ) {
+				self::$results[ $action ]['duplicate']++;
+				return;
+			}
+			\Newspack\Data_Events\Webhooks::handle_dispatch( $action, $timestamp, $data );
+			self::$results[ $action ]['processed']++;
+		}
+	}
+
+	/**
+	 * Backfill data for a specific action.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <action>
+	 * : The action to backfill. Choose "all" to backfill all actions.
+	 *
+	 * [--start=<start>]
+	 * : The start date for the backfill.
+	 *
+	 * [--end=<end>]
+	 * : The end date for the backfill.
+	 *
+	 * [--live]
+	 * : Run the backfill in live mode, which will process the events.
+	 *
+	 * [--verbose]
+	 * : Output more verbose information.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp newspack-network data-backfill reader_registered --start=2020-01-01 --end=2020-12-31 --live
+	 *
+	 * @param array $args The command arguments.
+	 * @param array $assoc_args The command options.
+	 * @return void
+	 */
+	public static function data_backfill( $args, $assoc_args ) { // phpcs:ignore Generic.NamingConventions.ConstructorName.OldStyle
+		$action = $args[0];
+		$start = isset( $assoc_args['start'] ) ? $assoc_args['start'] : null;
+		$end = isset( $assoc_args['end'] ) ? $assoc_args['end'] : null;
+		$live = isset( $assoc_args['live'] ) ? $assoc_args['live'] : null;
+		$verbose = isset( $assoc_args['verbose'] ) ? $assoc_args['verbose'] : null;
+
+		WP_CLI::line( '' );
+
+		if ( $action !== 'all' && ! in_array( $action, array_keys( Accepted_Actions::ACTIONS ) ) ) {
+			WP_CLI::error( 'Invalid action.' );
+		}
+
+		$is_hub = Site_Role::is_hub();
+		$is_node = Site_Role::is_node();
+		if ( ! $is_hub && ! $is_node ) {
+			WP_CLI::error( 'This command can only be run on a Hub or Node site.' );
+		}
+
+		if ( $is_hub ) {
+			WP_CLI::line( 'Running on a Hub site – will create events for the event log.' );
+		} else {
+			WP_CLI::line( 'Running on a Node site – will create webhook requests. This may result in duplicate requests if the request queue was cleared already, but the Hub will handle the duplicates.' );
+		}
+		WP_CLI::line( '' );
+
+		if ( ! method_exists( '\Newspack\Reader_Activation', 'get_reader_roles' ) ) {
+			WP_CLI::error( 'Incompatible Newspack plugin version.' );
+		}
+		if ( $live ) {
+			WP_CLI::line( '⚡️ Heads up! Running live, data will be updated.' );
```

Comment:
We could add a confirmation prompt here

Reply:
It already requires passing the `--live` flag explicitly, I don't see why it should be confirmed on top of that.